app/scripts/metamask-controller.js

```diff
@@ -2370,18 +2404,38 @@ export default class MetamaskController extends EventEmitter {
     return {
       isUnlocked: this.isUnlocked(),
       accounts: await this.getPermittedAccounts(origin),
-      ...this.getProviderNetworkState(),
+      ...this.getProviderNetworkState(
+        this.preferencesController.useRequestQueue() ? origin : undefined,
+      ),
     };
   }
 
   /**
-   * Gets network state relevant for external providers.
+   * Retrieves network state information relevant for external providers.
    *
-   * @returns {object} An object with relevant network state properties.
+   * @param {string} origin - The origin identifier for which network state is requested (default: 'metamask').
+   * @returns {object} An object containing important network state properties, including chainId and networkVersion.
    */
-  getProviderNetworkState() {
+  getProviderNetworkState(origin = 'metamask') {
+    let chainId;
+    if (this.preferencesController.getUseRequestQueue()) {
+      const networkClientId =
+        this.controllerMessenger.call(
+          'SelectedNetworkController:getNetworkClientIdForDomain',
+          origin,
+        ) || this.networkController.state.selectedNetworkClientId;
```

Comment:
determining default kind of overlaps with the SelectedNetworkMiddleware a little bit. would

```
        this.controllerMessenger.call(
          'SelectedNetworkController:getNetworkClientIdForDomain',
          origin,
        ) || this.controllerMessenger.call(
          'SelectedNetworkController:getNetworkClientIdForDomain',
          'metamask',
        )
 ```
 
 work?
 
 This pushes the concern of what fallback networkClientId to use back to the SelectedNetworkController

Reply:
hmm yeah. ATM im thinking it would be ncie to have selectedNetworkController always return a value, and have it decide how to default the values (in all cases we want the default to be 'what ever the globally  selected network is'). I ended up adding thiis defaulting here because of an issue where the selected network for the origin 'metamask' is undefined until you unlock the wallet and select a network for the first time.

Reply:
yeah... not in the scope of this PR to fix that. okay. May be worth leaving a comment explaining why this default logic is needed. Nit. Feel free to address or ignore and then mark this as resolved